conda_smithy/templates/travis.yml.tmpl

```diff
@@ -1,76 +1,117 @@
 # This file was generated automatically from conda-smithy. To update this configuration,
 # update the conda-forge.yml and/or the recipe/meta.yaml.
 # -*- mode: yaml -*-
-
+{%- set platformset = [] %}
+{%- for _, platform, _ in configs %}
+  {%- set pfarchless = platform.split('-')[0] %}
+  {%- if pfarchless not in platformset %}
+    {%- do platformset.append(pfarchless) %}
+  {%- endif %}
+{%- endfor %}
 language: generic
 
-os: osx
-osx_image: {{ travis.get('osx_image', 'xcode9.4') }}
-
-
-{% block env -%}
-{% if configs[0] or travis.secure -%}
+{% if travis.secure -%}
 env:
-  {%- if configs[0] %}
-  matrix:
-    {%- for config, _, _ in configs %}
-    - CONFIG={{ config }}
-    {%- endfor %}
-  {%- endif %}
-{##}
-  {%- if travis.secure %}
   global:
     {%- for name, hashed_secure in travis.secure | dictsort %}
     # The {{ name }} secure variable. This is defined canonically in conda-forge.yml.
     - secure: "{{ hashed_secure }}"
     {%- endfor %}
-  {%- endif %}
 {%- endif %}
-{% endblock %}
+
+{% if configs[0] -%}
+matrix:
+  include:
+    {%- for config, platform, upload in configs | sort %}
+    - env: CONFIG={{ config }} UPLOAD_PACKAGES={{ upload }}
```

Comment:
`UPLOAD_PACKAGES` is needed for CI that we add forcibly, which is only azure. I would remove this as it's not needed.

Reply:
Having the upload packages pattern be common to all the build systems would be quite handy imo.